jerry-ext/include/jerryscript-ext/arg.impl.h

```diff
@@ -17,42 +17,83 @@
 #define JERRYX_ARG_IMPL_H
 
 /* transform functions for each type. */
-jerry_value_t jerryx_arg_transform_number_strict (jerryx_arg_js_iterator_t *js_arg_iter_p,
-                                                  const jerryx_arg_t *c_arg_p);
-jerry_value_t jerryx_arg_transform_number_strict_optional (jerryx_arg_js_iterator_t *js_arg_iter_p,
-                                                           const jerryx_arg_t *c_arg_p);
-jerry_value_t jerryx_arg_transform_number_optional (jerryx_arg_js_iterator_t *js_arg_iter_p,
-                                                    const jerryx_arg_t *c_arg_p);
-jerry_value_t jerryx_arg_transform_number (jerryx_arg_js_iterator_t *js_arg_iter_p,
-                                           const jerryx_arg_t *c_arg_p);
-jerry_value_t jerryx_arg_transform_boolean_strict (jerryx_arg_js_iterator_t *js_arg_iter_p,
-                                                   const jerryx_arg_t *c_arg_p);
-jerry_value_t jerryx_arg_transform_boolean_strict_optional (jerryx_arg_js_iterator_t *js_arg_iter_p,
-                                                            const jerryx_arg_t *c_arg_p);
-jerry_value_t jerryx_arg_transform_boolean_optional (jerryx_arg_js_iterator_t *js_arg_iter_p,
-                                                     const jerryx_arg_t *c_arg_p);
-jerry_value_t jerryx_arg_transform_boolean (jerryx_arg_js_iterator_t *js_arg_iter_p,
-                                            const jerryx_arg_t *c_arg_p);
-jerry_value_t jerryx_arg_transform_string_strict (jerryx_arg_js_iterator_t *js_arg_iter_p,
-                                                  const jerryx_arg_t *c_arg_p);
-jerry_value_t jerryx_arg_transform_string_strict_optional (jerryx_arg_js_iterator_t *js_arg_iter_p,
-                                                           const jerryx_arg_t *c_arg_p);
-jerry_value_t jerryx_arg_transform_string_optional (jerryx_arg_js_iterator_t *js_arg_iter_p,
-                                                    const jerryx_arg_t *c_arg_p);
-jerry_value_t jerryx_arg_transform_string (jerryx_arg_js_iterator_t *js_arg_iter_p,
-                                           const jerryx_arg_t *c_arg_p);
-jerry_value_t jerryx_arg_transform_function (jerryx_arg_js_iterator_t *js_arg_iter_p,
-                                             const jerryx_arg_t *c_arg_p);
-jerry_value_t jerryx_arg_transform_function_optional (jerryx_arg_js_iterator_t *js_arg_iter_p,
-                                                      const jerryx_arg_t *c_arg_p);
-jerry_value_t jerryx_arg_transform_native_pointer (jerryx_arg_js_iterator_t *js_arg_iter_p,
-                                                   const jerryx_arg_t *c_arg_p);
-jerry_value_t jerryx_arg_transform_native_pointer_optional (jerryx_arg_js_iterator_t *js_arg_iter_p,
-                                                            const jerryx_arg_t *c_arg_p);
+
+#define DECLARE_JERRYX_ARG_TRANSFORM_FUNC_WITH_OPTIONAL(type) \
```

Comment:
Here I use the prefix DECLARE_ to indicate it is function declaration.And I use GENERATE_ to indicate it is function impl./defination